packages/snjs/lib/services/settings_service/SettingsGateway.ts

```diff
@@ -0,0 +1,97 @@
+import { SettingPayload } from '@standardnotes/settings';
+import * as messages from '../api/messages';
+import { SettingsProvider } from './SettingsProvider';
+import {
+  DeleteSettingResponse,
+  GetSettingResponse,
+  ListSettingsResponse,
+  UpdateSettingResponse,
+  User,
+} from '../api/responses';
+import { UuidString } from '@Lib/types';
+
+interface SettingsAPI {
+  listSettings(userUuid: UuidString): Promise<ListSettingsResponse>;
+
+  updateSetting(
+    userUuid: UuidString,
+    settingName: string,
+    settingValue: any
+  ): Promise<UpdateSettingResponse>;
+
+  getSetting(
+    userUuid: UuidString,
+    settingName: string
+  ): Promise<GetSettingResponse>;
+
+  deleteSetting(
+    userUuid: UuidString,
+    settingName: string
+  ): Promise<DeleteSettingResponse>;
+}
+
+/**
+ * SettingsGateway coordinates communication with the API service
+ * and transforms settings' payloads between ready to use objects
+ * and serialized values for the backend
+ */
+export class SettingsGateway implements SettingsProvider {
+  constructor(
+    private readonly api: SettingsAPI,
+    private readonly getUser: () => User | undefined
+  ) {}
+
+  isReadyForModification() {
+    return this.getUser() != null;
+  }
+
+  private get userUuid() {
+    const user = this.getUser();
+    if (user == null || user.uuid == null) {
+      throw new Error(messages.API_MESSAGE_INVALID_SESSION);
+    }
+    return user.uuid;
+  }
+
+  async listSettings() {
+    const { error, data } = await this.api.listSettings(this.userUuid);
+    if (error != null) throw new Error(error.message);
+    if (data == null || data.settings == null) return {};
+
+    const res: Partial<SettingPayload> = {};
```

Comment:
What does res stand for?

Reply:
Changed it to settings. It's the whole list of settings.